react-app/src/components/Todo.tsx

```diff
@@ -0,0 +1,169 @@
+import { useState, useEffect } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import {formatDistance} from "date-fns"
+import { ArrowDownAZ, ArrowDownZA, ListChecks, Plus, Search, XSquare} from "lucide-react"
+import { ITodo } from "../types/ITodo";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import axios from 'axios';
+import {useReducer} from 'react'
+
+const initialValues = {
+  search : '',
+  page : 1,
+  limit : 3,
+  order : 'asc',
+  todoStatus : '',
+  totalPages : 1
+}
+
+const reducer = (currentState:any, action:any)=>{
+  switch(action.type){
+    case "SET_ORDER" : return {...currentState, order: action.value};
+    case 'SET_PAGE' : return {...currentState, page: action.value};
+    case 'SET_LIMIT' : return {...currentState, limit: action.value};
+    case'SET_SEARCH' : return {...currentState, search: action.value};
+    case 'SET_TODOSTATUS' : return {...currentState, todoStatus: action.value};
+    case 'SET_TOTALPAGES' : return {...currentState, totalPages: action.value};
+    default : return currentState;
+  }
+}
+
+const Todo = () => {
+  // todo data
+  const [state, dispatch] = useReducer(reducer, initialValues);
+  const [todos, setTodos] = useState<ITodo[]>([]);
+  const navigate = useNavigate();
+  const {data, error, isPending} = useQuery({
+    queryKey: ['todos', state.page, state.limit, state.search, state.order, state.todoStatus],
+    queryFn: async() => {
+        const res = await axios.get(`http://localhost:5000/todos?_page=${state.page}&_limit=${state.limit}&title_like=${state.search}&_sort=title&_order=${state.order}&isCompleted_like=${state.todoStatus}`);
```

Comment:
please use prettier to format code.

also use helper method to build query parameters